src/twisted/newsfragments/7945.feature

```diff
@@ -0,0 +1 @@
+Twisted's cross-platform C extensions (mainly used for twisted.python.sendmsg) are now distributed in a separate package, removing the requirement for a C compiler on Linux/MacOS/FreeBSD for the majority of Twisted's functionality. This new package can be installed by ``pip install Twisted[c]``.
```

Comment:
I really don't like `[c]` as a name for this.

setuptools extras are already inscrutable enough without having their names refer to random implementation details (in this case, that the functionality in question happens to be written in C).  Many of Twisted's other extras require code written in C, but we're not calling it `twisted[libcrypto,libssl]` because the _reason_ you install those dependencies is `tls`.

In this case, the reason that you (as a user) want this code is that it makes `IUNIXTransport.sendFileDescriptor` work on Python 2.7.  So:

- Could we call the extra `sendfd27`?
- Could we update the docs for `IUNIXTransport.sendFileDescriptor`?
- I'm pretty sure this should be included in one of the aggregates, like `all_non_platform` .

(The fact that our own test suite needs it to test `failure` is a dumb implementation detail that we can keep secret, as far as the extras are concerned.)